src/exo/memory.py

```diff
@@ -58,6 +58,34 @@ class MemGenError(Exception):
     pass
 
 
+def generate_offset(indices, strides):
+    def index_expr(i, s):
+        if s == "0" or i == "0":
+            return ""
+
+        if s == "1":
+            return i
+        if i == "1":
+            return s
+
+        if len(s) == 1:
+            return f"({i}) * {s}"
+        else:
+            return f"({i}) * ({s})"
+
+    offset = index_expr(indices[0], strides[0])
```

Comment:
I think we can rewrite this to map all the indices, then filter out the "", and then either return `"+".join(exprs)` if `len(exprs) > 1` and `"0"` otherwise.

Reply:
True, will do that!